### PR TITLE
Add Broker Horizontal Up-scale and Broker Rolling Update when Name Service Scaling

### PR DESCRIPTION
**Pull request by liuruiyiyang**
Change List:
1. Add a new feature that the broker cluster can up-scale with metadata copied from a user-defined source broker pod for the out-of-order message scenario.
2. Implement the rolling updating of broker pods when name service cluster scaling.
3. Add documents related to the above features and a content table for reader navigation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -287,10 +307,45 @@ $ cat default-broker-storage-broker-1-master-0-pvc-d7b76efe-384c-4f8d-9e8a-ebe20
...
 ```
 
+## Horizontal Scale
+
+### Name Server Cluster Scale
+If the current name service cluster scale does not fit your requirements, you can simply use RocketMQ-Operator to up-scale or down-scale your name service cluster.
+
+If you want to enlarge your name service cluster. Modify your name service CR file ```rocketmq_v1alpha1_nameservice_cr.yaml```, increase the field ```size``` to the number you want, for example, from ```size: 1``` to ```size: 2```.
+
+> Notice: if your broker image version is 4.5.0 or earlier, you need to make sure that ```allowRestart: true``` is set in the broker CR file to enable rolling restart policy. If not, configure it to true and run ```kubectl apply -f example/rocketmq_v1alpha1_broker_cr.yaml``` to apply the new config.
```
Comment (vongosling) [inline README.md]: If not, configure it to false

File: pkg/constants/constants.go [hunk 2]
```diff
@@ -18,9 +18,10 @@
 package constants
 
 const BrokerClusterPrefix = "broker-cluster-"
-const MasterBrokerContainerNamePrefix = "broker-master"
-const ReplicaBrokerContainerNamePrefix = "broker-replica"
+const BrokerContainerName = "broker"
 const AdminToolDir = "/home/rocketmq/rocketmq-4.5.0/bin/mqadmin"
```
Comment (vongosling) [inline pkg/constants/constants.go]: can not use for 4.6.0 ？

File: pkg/controller/nameservice/nameservice_controller.go [hunk 3]
```diff
@@ -188,9 +188,9 @@ func (r *ReconcileNameService) updateNameServiceStatus(instance *rocketmqv1alpha
 		share.NameServersStr = nameServerListStr[:len(nameServerListStr)-1]
 		reqLogger.Info("share.NameServersStr:" + share.NameServersStr)
 
-		if len(oldNameServerListStr) < 2 {
+		if len(oldNameServerListStr) < 8 {
```
Comment (vongosling) [inline pkg/controller/nameservice/nameservice_controller.go]: magic number

File: README.md (hunk 1 above)
Comment (liuruiyiyang) [reply README.md]: This notice has been clarified.

File: pkg/constants/constants.go (hunk 2 above)
Comment (liuruiyiyang) [reply pkg/constants/constants.go]: We will change the admin tool directory name to a version-free name later.

File: pkg/controller/nameservice/nameservice_controller.go (hunk 3 above)
Comment (liuruiyiyang) [reply pkg/controller/nameservice/nameservice_controller.go]: We will replace this magic number with a meaningful constant.

**Pull request merged**